nanoFramework.Device.Bluetooth/BluetoothLEServer.cs

```diff
@@ -0,0 +1,267 @@
+
+        /// <summary>
+        /// Returns BluetoothLEServer singleton instance.
+        /// </summary>
+        public static BluetoothLEServer Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    lock (_lock)
+                    {
+                        if (_instance == null)
+                        {
+                            _instance = new BluetoothLEServer();
+                        }
+                    }
+                }
+                return _instance;
+            }
+        }
+
+        /// <summary>
+        /// Construct new Bluetooth Server using primary service UUID
+        /// </summary>
+        private BluetoothLEServer()
+        {
+            this.disposedValue = false;
+
+            Init();
+        }
+
+        private void Init()
+        {
+            _session = new GattSession(new BluetoothDeviceId(0));
+            _pairing = new DevicePairing(this);
+            _services = new();
+
+            _bluetoothEventManager.BluetoothServer = this;
+            _bluetoothEventManager.Reset();
+        }
+
+        /// <summary>
+        /// Server device name, defaults to 'nanoFramework'
+        /// </summary>
+        public String DeviceName { get => BluetoothNanoDevice.DeviceName; set => BluetoothNanoDevice.DeviceName = value; }
+
+        /// <summary>
+        /// Get GattSession aasoicated with this server.
+        /// </summary>
+        public GattSession Session { get => _session; }
+
+        #region Services
+
+        /// <summary>
+        /// Returns service provider with specifued UUID
+        /// </summary>
+        /// <param name="serviceUuid"></param>
+        /// <returns>return GattServiceProvider object or null if not found.</returns>
+        public GattServiceProvider GetServiceByUUID(Guid serviceUuid)
+        {
+            foreach (GattServiceProvider srvprov in _services)
+            {
+                if (srvprov.Service.Uuid.Equals(serviceUuid))
+                {
+                    return srvprov;
+                }
+            };
+
+            return null;
+        }
+
+        /// <summary>
+        /// Get an array of all associated services providers for this Bluetooth LE Server.
+        /// </summary>
+        /// <remarks>
+        /// The primary service will be index 0 followed by the Device Information at index 1.
+        /// Any other Services added to server will follow these in the order they were created.
+        /// </remarks>
+        public GattServiceProvider[] Services { get { return (GattServiceProvider[])_services.ToArray(typeof(GattServiceProvider)); } }
+
+        #endregion
+
+        #region Security/Pairing
+
+        /// <summary>
+        /// Return Pairing object for handling Pairing.
+        /// </summary>
+        public DevicePairing Pairing { get => _pairing; }
+
+        /// <summary>
+        /// Evaulate all services and work out security requirements
```

Comment:
```suggestion
        /// Evaulates all services and work out security requirements.
```

Reply:
This has been fixed but method probably needs to be removed as duplicate